### PR TITLE
STSMACOM-957 Handle undefined response in processBadResponse

### PR DESCRIPTION
**Pull request by MikeTaylor**
Add guard clause to handle undefined response in processBadResponse function.

I ran across this while developing ui-cyclops, and running against a server that had not yet had the new permissions inserted. Obviously that's a bit of an edge-case as it's unlikely to happen on a well-managed production system. But I do think that no permissions state should ever cause an actual crash.

New jest tests cover this condition; existing BT suite allowed it to slip through.

Refs [STSMACOM-957](https://folio-org.atlassian.net/browse/STSMACOM-957)

### CODE REVIEW

File: lib/EditableList/processBadResponse.js [hunk 1]
```diff
@@ -68,6 +68,16 @@ const getDefaultErrorMessage = (defaultErrorCode) => {
...
 export default async function processBadResponse(response, defaultErrorCode, getCustomErrorMessages = null) {
+  // ControlledVocab rejects its delete promise with no
+  // argument when the DELETE fails (e.g. a 403), so `response` arrives here as
+  // undefined and `response.status` throws, crashing the whole app instead of
+  // surfacing the failure. Guard against a non-Response reason and fall back to
+  // the default error message. See stripes-smart-components ControlledVocab
+  // onDeleteItem -> deleteItemReject().
+  if (!response || typeof response.status !== 'number') {
+    return getDefaultErrorMessage(defaultErrorCode);
+  }
```
Comment (zburke) [inline lib/EditableList/processBadResponse.js]: Alternatively, instead of adding new condition we change existing conditions that inspect `response.status`:
```
if (response.status === 422) {  // throws if response is undefined
if (response?.status === 422) { // keeps calm and carries on
```
or change the overall structure to only handle the default in one place:
```
if (response?.status) {
  if (response.status === 422) ...
  if (response.status === 409) ...
}
return getDefaultErrorMessage(defaultErrorCode);
```

**Review by zburke — APPROVED** [review]
Thanks for this. I appreciate the code-comments that explain why this error handling is necessary, and I *really appreciate that you took the time to fix it rather than just flagging it and leaving it for somebody else to deal with. 

The current implementation resolves the problem, but now handles the default/unknown-status case in two separate places. Consider the alternatives that only handle it once. 

I filed a corresponding ticket (assigned to you) and added it to the PR title. I hope this doesn't feel meddlesome. It provides visibility of this change for POs and makes it easier for them to track which release contains a given bug-fix.

File: lib/EditableList/processBadResponse.js (hunk 1 above)
Comment (MikeTaylor) [reply lib/EditableList/processBadResponse.js]: I'm happy with any of these approaches.

Comment (zburke) [reply lib/EditableList/processBadResponse.js]: Do you want to pick your favorite and merge it, or do you want somebody else to pick up this work and merge it? I trust your judgement and have approved the PR to avoid blocking your progress.

Comment (MikeTaylor) [reply lib/EditableList/processBadResponse.js]: Thanks. I think I'd prefer you to make the change you like best. It's not a showstopper for me, so if you need another day or two to think it through, that won't slow me down.
